src/v2i-hub/CARMAStreetsPlugin/src/J2735MapToJsonConverter.cpp

```diff
@@ -0,0 +1,267 @@
+                intersectionJson["ref_point"]["elevation"] = std::to_string(*intersection->refPoint.elevation);
+
+                // Convert Laneset
+                Json::Value laneSetJson;
+                convertLanesetToJSON(intersection, laneSetJson);
+                intersectionJson["lane_set"] = laneSetJson;
+            }
+            intersectionsJson["intersection_geometry"] = intersectionJson;
+        }
+        mapDataJson["intersections"] = intersectionsJson;
+        mapJson["map_data"] = mapDataJson;
+    }
+
+    void J2735MapToJsonConverter::convertLanesetToJSON(const IntersectionGeometry *intersection, Json::Value &laneSetJson) const
+    {
+        // Construct laneset
+        const auto laneSet = intersection->laneSet;
+        if (laneSet.list.array != nullptr)
+        {
+            for (size_t i = 0; i < laneSet.list.count; i++)
+            {
+                std::stringstream ss;
+                Json::Value laneJson;
+                auto lane = laneSet.list.array[i];
+                laneJson["lane_id"] = std::to_string(lane->laneID);
+                if (lane->ingressApproach != nullptr)
+                {
+                    ss.str("");
+                    ss << *lane->ingressApproach;
+                    laneJson["ingress_approach"] = ss.str();
+                }
+                if (lane->egressApproach != nullptr)
+                {
+                    ss.str("");
+                    ss << *lane->egressApproach;
+                    laneJson["egressApproach"] = ss.str();
+                }
+
+                // Construct LaneAttributes
+                Json::Value LaneAttributesJson;
+                convertLaneAttributeToJSON(lane, LaneAttributesJson);
+                laneJson["lane_attributes"] = LaneAttributesJson;
+
+                // Construct nodelist
+                Json::Value nodeList;
+                convertNodeListToJSON(lane, nodeList);
+                laneJson["node_list"]["nodes"] = nodeList;
+
+                // Construct connects
+                Json::Value connectsJson;
+                if (lane->connectsTo != nullptr)
+                {
+                    convertConnectsToJSON(lane, connectsJson);
+                    laneJson["connects_to"] = connectsJson;
+                }
+
+                laneSetJson.append(laneJson);
+            }
+        }
+    }
+
+    void J2735MapToJsonConverter::convertLaneAttributeToJSON(const GenericLane *lane, Json::Value &LaneAttributesJson) const
+    {
+        std::stringstream ss;
+        ss.str("");
+        auto bit_unused = lane->laneAttributes.directionalUse.bits_unused;
+        ss << lane->laneAttributes.directionalUse.buf;
+        auto binary = lane->laneAttributes.directionalUse.buf[0] >> bit_unused;
+        std::string binary_str = std::to_string(static_cast<unsigned>(binary / 2)) + std::to_string(static_cast<unsigned>(binary % 2));
```

Comment:
What conversion is happening here?

Reply:
directionalUse is one byte. We are sending binary 01 or 10 in a string format. I can send you the XML.